lib/grax/schema/registry.ex

```diff
@@ -28,7 +28,7 @@ defmodule Grax.Schema.Registry do
 
   @impl true
   def init(opts) do
-    {:ok, State.build(opts)}
+    {:ok, State.cached() || State.build(opts)}
   end
 
```

Comment:
I think, things would get much simpler by using just an internal protocol for tracking modules using Grax
example here https://github.com/semarco/poc_compile_time_module_usage_tracking/blob/main/lib/tracker.ex